### PR TITLE
642 face cam

### PR DESCRIPTION
**Pull request by koschke**
This branch is workings towards showing the video of all desktop player on all clients. Currently that is not the case as reported in issue #642. The problem is not yet fixed but in the course of starting to work on this issue here in this branch I fixed a null-pointer dereference during start-up of a player's avatar, reported by @falko17.

The changes made in this pull request fixes the problem and should be merged into the master. The branch will continue to exist for the additional work required for issue #642.

### CODE REVIEW

File: Assets/SEE/Tools/FaceCam/FaceCam.cs [hunk 1]
```diff
@@ -253,6 +257,36 @@ public override void OnNetworkSpawn()
...
+                if (!gameObject.TryGetComponentOrLog(out webCamTextureToMatHelper))
+                {
+                    enabled = false;
+                }
+            }
+
+            if (!gameObject.TryGetComponentOrLog(out meshRenderer))
+            {
+                gameObject.SetActive(false);
+                return;
```
Comment (falko17) [inline Assets/SEE/Tools/FaceCam/FaceCam.cs:286]: This line seems unnecessary.
```suggestion
```

File: Assets/SEE/Game/LabelAttributes.cs [hunk 2]
```diff
@@ -24,13 +34,13 @@ public class LabelAttributes
         /// The distance between the top of the node and its label.
         /// </summary>
         [Tooltip("The distance between the top of the node and its label.")]
-        public float Distance = 0.2f;
+        public float Distance = DEFAULT_DISTANCE;
```
Comment (koschke) [inline Assets/SEE/Game/LabelAttributes.cs:37]: Names must be in CamelCase.
Private fields must start with a lower-case letter, protected, internal and public fields with upper-case letter. This holds for constant and readonly declarations, too.

File: Assets/SEE/Tools/FaceCam/FaceCam.cs [hunk 3]
```diff
@@ -671,6 +735,8 @@ private void FaceCamOnFrontToggleServerRpc(bool networkFaceCamOnFront)
         [ClientRpc]
         private void FaceCamOnFrontToggleClientRpc(bool networkFaceCamOnFront)
         {
+            Debug.Log($"[RPC] Client {NetworkManager.Singleton.LocalClientId} received FaceCamOnFrontToggleClientRpc from server with networkFaceCamOnFront={networkFaceCamOnFront}\n");
```
Comment (falko17) [inline Assets/SEE/Tools/FaceCam/FaceCam.cs:738]: Just a thought: There's a lot of these log outputs which are useful for debugging but aren't really needed by default. Maybe we can gate these behind a debug flag? Or are these just temporary until the FaceCam issues are resolved?

File: Assets/SEE/Game/LabelAttributes.cs [hunk 4]
```diff
@@ -24,13 +34,13 @@ public class LabelAttributes
...
 
         /// <summary>
         /// The font size of the node's label.
         /// </summary>
         [Tooltip("The font size of the label.")]
-        public float FontSize = 0.4f;
+        public float FontSize = DEFAULT_FONT_SIZE;
```
Comment (koschke) [inline Assets/SEE/Game/LabelAttributes.cs:43]: Names must be in CamelCase.
Private fields must start with a lower-case letter, protected, internal and public fields with upper-case letter. This holds for constant and readonly declarations, too.

File: Assets/SEE/Game/LabelAttributes.cs (hunk 2 above)
Comment (falko17) [reply Assets/SEE/Game/LabelAttributes.cs]: Fixed in abf328e4b12dd2f3276416e37bbe466f2e306a5b.

File: Assets/SEE/Game/LabelAttributes.cs (hunk 4 above)
Comment (falko17) [reply Assets/SEE/Game/LabelAttributes.cs]: Fixed in abf328e4b12dd2f3276416e37bbe466f2e306a5b.

**Review by koschke — COMMENTED** [review]
There are two violations of our naming rules.

File: Assets/SEE/Tools/FaceCam/FaceCam.cs (hunk 3 above)
Comment (koschke) [reply Assets/SEE/Tools/FaceCam/FaceCam.cs:738]: These will be removed when the issue is resolved.

Comment (koschke) [reply Assets/SEE/Tools/FaceCam/FaceCam.cs:745]: I added #ifdefs.

File: Assets/SEE/Game/Worlds/PlayerSpawner.cs [hunk 5]
```diff
@@ -110,27 +105,50 @@ private void Spawn(ulong owner)
...
+                //
+                // (1) the same GameObject that the NetworkObject component is attached to
+                // (2) any child or children of the GameObject that the NetworkObject is attached to.
+                //
+                // A caveat of the above two rules is when one of the children GameObjects also
+                // has a NetworkObject component assigned to it (a.k.a. "Nested NetworkObjects").
+                // Nested NetworkObject components aren't permited in network prefabs.
+
+                //GameObject faceCam = PrefabInstantiator.InstantiatePrefab("Prefabs/FaceCam/FaceCam", parent: player.transform);
+#if false // FIXME
```
[the remark this answers is not in the document]
Comment (koschke) [reply Assets/SEE/Game/Worlds/PlayerSpawner.cs]: I added a comment.

File: Assets/SEE/Tools/FaceCam/FaceCam.cs (hunk 1 above)
Comment (koschke) [reply Assets/SEE/Tools/FaceCam/FaceCam.cs:286]: True, it's not needed, but what if anyone adds a line after the if statement at a later point in time?

File: Assets/SEE/Game/LocalPlayer.cs [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+    /// </summary>
+    public static class LocalPlayer
+    {
+        /// <summary>
+        /// The game object representing the active local player, that is, the player
+        /// executing on this local instance of Unity.
+        /// </summary>
+        public static GameObject Instance;
+    }
+}
```
[the remark this answers is not in the document]
Comment (koschke) [reply Assets/SEE/Game/LocalPlayer.cs]: I added a newline.

**Review by falko17 — APPROVED** [review]
Should be good to merge once the pipeline runs through.

**Pull request merged**